core/storage/btree.rs

```diff
@@ -5560,8 +5693,9 @@ impl CursorTrait for BTreeCursor {
                                 ..
                             }) => {
                                 self.stack.advance();
-                                let (mem_page, c) = self.read_page(left_child_page as i64)?;
-                                self.stack.push(mem_page);
+                                let (child, c) =
```

Comment:
How does this work? If we yield on IO here, then `child` won't be pushed to stack. If we don't yield on IO here, then is `c` ever `Some`?